src/gstring.rs

```diff
@@ -0,0 +1,333 @@
+        self.as_str()
+    }
+}
+
+impl Ord for GString {
+    fn cmp(&self, other: &GString) -> Ordering {
+        self.as_str().bytes().cmp(other.as_str().bytes())
+    }
+}
+
+impl PartialOrd for GString {
+    fn partial_cmp(&self, other: &GString) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for GString {
+    fn eq(&self, other: &GString) -> bool {
+        self.as_str() == other.as_str()
+    }
+}
+
+impl PartialEq<GString> for std::string::String {
+    fn eq(&self, other: &GString) -> bool {
+        self.as_str() == other.as_str()
+    }
+}
+
+impl PartialEq<str> for GString {
+    fn eq(&self, other: &str) -> bool {
+        self.as_str() == other
+    }
+}
+
+impl PartialEq<&'static str> for GString {
+    fn eq(&self, other: &&'static str) -> bool {
+        self.as_str() == *other
+    }
+}
+
+impl Eq for GString {}
+
+impl AsRef<str> for GString {
+    fn as_ref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl Deref for GString {
+    type Target = str;
+
+    fn deref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl From<GString> for String {
+    #[inline]
+    fn from(holder: GString) -> Self {
+        String::from(holder.as_str())
+    }
+}
+
+impl From<GString> for Box<str> {
+    #[inline]
+    fn from(holder: GString) -> Self {
+        Box::from(holder.as_str())
+    }
+}
+
+impl From<&'static str> for GString {
+    #[inline]
+    fn from(s: &'static str) -> Self {
+        let ptr = s.as_bytes();
+        GString::from(ptr)
+    }
+}
+
+impl FromGlibPtrFull<*const c_char> for GString {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *const c_char) -> Self {
+        GString::new(ptr)
+    }
+}
+
+impl FromGlibPtrFull<*mut u8> for GString {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut u8) -> Self {
+        GString::new(ptr as *const c_char)
+    }
+}
+
+impl FromGlibPtrFull<*mut i8> for GString {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut i8) -> Self {
+        GString::new(ptr as *const c_char)
+    }
+}
+
+impl FromGlibPtrNone<*const c_char> for GString {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *const c_char) -> Self {
+        let cstr = CStr::from_ptr(ptr);
+        GString::from_cstring(cstr.to_owned())
+    }
+}
+
+impl FromGlibPtrNone<*mut u8> for GString {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut u8) -> Self {
+        let cstr = CStr::from_ptr(ptr as *mut i8);
+        GString::from_cstring(cstr.to_owned())
+    }
+}
+
+impl FromGlibPtrNone<*mut i8> for GString {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut i8) -> Self {
+        let cstr = CStr::from_ptr(ptr);
+        GString::from_cstring(cstr.to_owned())
+    }
+}
+
+impl<'a> ToGlibPtr<'a, *const c_char> for GString {
+    type Storage = Option<CString>;
+
+    #[inline]
+    fn to_glib_none(&'a self) -> Stash<'a, *const c_char, Self> {
+        let cstring = self.to_cstring();
+        Stash(cstring.as_ptr(), Some(cstring))
+    }
```

Comment:
@philn, @sdroege Maybe this implementation can be simplified:
```rust
impl<'a> ToGlibPtr<'a, *const c_char> for GString {
    type Storage = &'a Self;
    #[inline]
    fn to_glib_none(&'a self) -> Stash<'a, *const c_char, Self> {
        Stash(self.as_ptr() as *const _, self)
    }
```

Reply:
Yes!